jgrapht-core/src/main/java/org/jgrapht/experimental/dag/DirectedAcyclicGraph.java

```diff
@@ -532,6 +533,82 @@ private void reorder(Set<V> df, Set<V> db, Visited visited)
     }
 
     /**
+     * This function is implemented recursively. Therefore only graphs without
+     * any cycles are allowed.
+     *
+     * @param graph graph to look for ancestors in.
+     * @param vertex the vertex to get the ancestors of.
+     *
+     * @return {@link List} of ancestors of the vertex in the given graph.
+     */
+    public static <V, E extends DefaultEdge> List<V> getAncestors(
+        DirectedAcyclicGraph<V, E> graph,
+        V vertex)
+    {
+
+        List<V> ancestors = new ArrayList<>();
+        ancestors = getAncestors(graph, vertex, ancestors);
+        return ancestors;
+    }
+
+    protected static <V, E extends DefaultEdge> List<V> getAncestors(
```

Comment:
I am now using iterators instead of recursion to determine the descandants/ancestors. If I would have known that determining the ancestors is as easy as reversing the graph using `EdgeReversedGraph` and iterating over it using the `DepthFirstIterator ` I would have done this in the first place :-) See commit _6d9a91c81918b2f05c8a6b8cdf1f6e41592b9d6b_.